### DISCUSSION TITLE
Span.fromContext(Context.current()).getSpanContext() return PropagatedSpan.INVALID

### DISCUSSION THREAD

**Discussion by cleverchuk** [open-telemetry/opentelemetry-java-instrumentation#7728] [Q&A] [answered]
After upgrading our embedded extensions to version 1.22.0 of the agent, the instrumentation broke and span is consistently invalid. Please, I need help figuring out why the api is returning PropagatedSpan.INVALID when called.
I'm a noob in this stuff so a lot things aren't obvious to me.

Additional Details:

- instrumented library: servlet 3
- Servlet container: Tomcat
- Span.fromContext(Context.current()).getSpanContext() return PropagatedSpan.INVALID

**Comment by laurit** [comment]
You could enable debug logging with `-Dotel.javaagent.debug=true` and see if there is anything suspicious. It would help if you could provide a minimal sample that reproduces your issue.

**Comment by cleverchuk** [comment, marked as answer]
I was able to solve the problem. By using similar configuration as is found [here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/distro) and ensuring that any dependency on OTEL is `compileOnly`. This ensures that `bootstrap` class loader and the `agent` class loader don't end up loading the same exact class. I also refactored the project so it uses same structure as the one linked.

I believe the root cause was the project was using the `buildSrc` approach to repackage the jar.
